STM32Cube/Tasks/flight_phase.c

```diff
@@ -8,53 +8,113 @@
 #include "flight_phase.h"
 #include "millis.h"
 #include "printf.h"
+#include "log.h"
+#include "otits.h"
 
 #define BOOST_TIME_MS 10000 //time from inj valve open to motor bunout
 #define RECOVERY_DELAY_MS 15000 //time from inj valve open to recovery deployment
 
-enum FLIGHT_PHASE flight_state = PHASE_PRELAUNCH;
-float injOpenTime;
-float deploymentTime;
+static FLIGHT_PHASE flight_state = PHASE_PRELAUNCH;
+static float injOpenTime;
+static float deploymentTime;
 
 EventGroupHandle_t flightPhaseEventsHandle = NULL;
 #define INJ_OPEN (xEventGroupGetBits(flightPhaseEventsHandle) & INJ_OPEN_BIT)
 #define RECOVERY_DEPLOYMENT (xEventGroupGetBits(flightPhaseEventsHandle) & RECOVERY_DEPLOYMENT_BIT)
 
+static Otits_Result_t test_flightPhase() {
+    Otits_Result_t res;
+    float time = millis_();
 
-void flightPhaseInit()
-{
+    // check that millis matches tick count to a reasonable degree TODO: is this how timer works lmao
+    if (time - xTaskGetTickCount() > 2) {
```

Comment:
is this interpreting the function of millis_() correctly

Reply:
Yes. I mean technically if you change the timebase of freertos it isn't, but I think its fair to assume we use 1000 ticks/second